python/cppyy/numba_ext.py

```diff
@@ -56,28 +59,50 @@ class Qualified:
     'unsigned long long'     : nb_types.ulonglong,
     'float'                  : nb_types.float32,
     'double'                 : nb_types.float64,
+    'const char*'           : nb_types.unicode_type,
 }
 
 def cpp2numba(val):
     if type(val) != str:
         # TODO: distinguish ptr/ref/byval
         return typeof_scope(val, nb_typing.typeof.Purpose.argument, Qualified.value)
+    elif val[-1] == '*':
+        return nb_types.CPointer(_cpp2numba[val[:-1]])
     return _cpp2numba[val]
 
 _numba2cpp = dict()
 for key, value in _cpp2numba.items():
-    _numba2cpp[value] = key
-# prefer "int" in the case of intc over "int32_t"
-_numba2cpp[nb_types.intc] = 'int'
+    if value in _numba2cpp:
+        _numba2cpp[value].append(key)
+        continue
+    _numba2cpp[value] = [key]
+
+
+def refactor_eigen_to_cpp(input_string):
```

Comment:
This shouldn't live in the general `numba_ext.py`. Eigen-specific support is fine, but it should live in its own module that is only imported for folks interested in Eigen.